cloudapp/src/app/user.service.ts

```diff
@@ -62,15 +66,17 @@ export class UserService {
 
     private extractUserFromAlmaUser = (almaUser, index) => {
         return {
-            id: this.usersRowNumber[index],
+            id: index, //this.usersRowNumber[index],
             name: almaUser.full_name.search('null ') === 0 ? almaUser.full_name.replace('null ', '') : almaUser.full_name,
             addresses: almaUser.contact_info.address.map(
                 address => ({
                     type: address.address_type[0].value,
                     address: this.convertToPrintableAddress(address)
                 })
             ),
-            selectedAddress: almaUser.contact_info.address.find(address => address.preferred == true).address_type[0].value,
+            selectedAddress: almaUser.contact_info.address.some(address => address.preferred == true) 
+                ? almaUser.contact_info.address.find(address => address.preferred == true).address_type[0].value 
+                : almaUser.contact_info.address.length > 0 ? almaUser.contact_info.address[0].address_type[0].value : 'none',
```

Comment:
This is the main change which will prevent an error if the addresses for a user aren't available or a preferred address is not selected.